include/net/net_if.h

```diff
@@ -461,7 +461,7 @@ struct net_if_dev {
  * There can be many net_if instance against the same net_if_dev instance.
  *
  */
-struct net_if {
+__net_iface struct net_if {
```

Comment:
There is just one network interface struct, `struct net_if`. I'm not seeing a need here for `__net_iface`, I would prefer if you just added "net_if" to the `kobjects` dictionary in `gen_kobject_list.py`

Are you expecting more objects of this type?

Reply:
> Are you expecting more objects of this type?

Not sure. There is a net_if_dev pointer inside net_if so we need to allow access to it, we could mark it too as __net_iface. But that can be done later if we really need it.

Reply:
Yeah I would revisit this later if we decided we needed, this kind of grouping only works if the objects are interchangeable since there will be no way to distinguish them for validation purposes.